src/app/components/dynamicdialog/dialogservice.ts

```diff
@@ -17,6 +18,15 @@ export class DialogService {
 
         this.dialogComponentRefMap.get(dialogRef).instance.childComponentType = componentType;
 
+        const parentDialog = this.injector.get(DynamicDialogComponent, null);
+        if (parentDialog) {
+            parentDialog.unbindDocumentKeydownListener();
+            dialogRef.onDestroy.pipe(
+                take(1),
+                tap(() => parentDialog.bindDocumentKeydownListener())
```

Comment:
Hi, Why use tap instead of logic inside subscribe?

Reply:
Just a personnal preference, based on a lot of code reviewing :

*   easier to see the chain of treatment for the observable
*   if the subscribe need to be extracted (for example to be used with an async pipe), the refactor is easy
*   the subscribe accept in fact 3 arguments (next, error and complete handler), so always using it with no argument reduce a misplacement for those handlers

I could resume this way of writing my code like 'declare ; prepare ; execute' 😅

Reply:
The current project uses subscribe instead of `tap` more often. `Tap` also has 3 arguments)

Reply:
OK that make sense (consistency is key) ;-)